### ISSUE TITLE
Incorrect trailing comments

### ISSUE THREAD

**Issue by nzakas** [eslint/espree#212]
Given this code:

```js
//foo
var zzz /*aaa*/ = 777;
//bar
```

Espree incorrectly appends `//foo` as a trailing comment on the `Literal` 777.

**Comment by kaicataldo** [comment]
I'm going to take a look at this now that the leading comments bug is fixed

**Comment by kaicataldo** [comment]
This doesn't seem to be an issue any more. However, wanted to confirm that the current behavior of leading comments is correct. Namely, wondering if `//foo` should be appending to the leading comments of the Literal 777.

```js
{
    "type": "Program",
    "start": 0,
    "end": 34,
    "range": [
        6,
        28
    ],
    "body": [
        {
            "type": "VariableDeclaration",
            "start": 6,
            "end": 28,
            "range": [
                6,
                28
            ],
            "declarations": [
                {
                    "type": "VariableDeclarator",
                    "start": 10,
                    "end": 27,
                    "range": [
                        10,
                        27
                    ],
                    "id": {
                        "type": "Identifier",
                        "start": 10,
                        "end": 13,
                        "range": [
                            10,
                            13
                        ],
                        "name": "zzz",
                        "trailingComments": [
                            {
                                "type": "Block",
                                "value": "aaa",
                                "start": 14,
                                "end": 21,
                                "range": [
                                    14,
                                    21
                                ]
                            }
                        ]
                    },
                    "init": {
                        "type": "Literal",
                        "start": 24,
                        "end": 27,
                        "range": [
                            24,
                            27
                        ],
                        "value": 777,
                        "raw": "777",
                        "leadingComments": [
                            {
                                "type": "Line",
                                "value": "foo",
                                "start": 0,
                                "end": 5,
                                "range": [
                                    0,
                                    5
                                ]
                            },
                            {
                                "type": "Block",
                                "value": "aaa",
                                "start": 14,
                                "end": 21,
                                "range": [
                                    14,
                                    21
                                ]
                            }
                        ]
                    }
                }
            ],
            "kind": "var",
            "leadingComments": [
                {
                    "type": "Line",
                    "value": "foo",
                    "start": 0,
                    "end": 5,
                    "range": [
                        0,
                        5
                    ]
                }
            ],
            "trailingComments": [
                {
                    "type": "Line",
                    "value": "bar",
                    "start": 29,
                    "end": 34,
                    "range": [
                        29,
                        34
                    ]
                }
            ]
        }
    ],
    "sourceType": "script",
    "comments": [
        {
            "type": "Line",
            "value": "foo",
            "start": 0,
            "end": 5,
            "range": [
                0,
                5
            ]
        },
        {
            "type": "Block",
            "value": "aaa",
            "start": 14,
            "end": 21,
            "range": [
                14,
                21
            ]
        },
        {
            "type": "Line",
            "value": "bar",
            "start": 29,
            "end": 34,
            "range": [
                29,
                34
            ]
        }
    ]
}
```

**Comment by nzakas** [comment]
`//foo` should be in the leading comments of the VariableDeclaration only.

**Comment by kaicataldo** [comment]
Ah, thanks, that's what I thought. Looks like I have some more work to do. Not sure how to accomplish this one, but will spend some time with it and see what I come up with

**Comment by kaicataldo** [comment]
I wonder if we should close this and open a new issue for the duplicate leading comment not being removed from the literal `777` or maybe just rename this issue? The original bug the issue is reporting seems to me to be fixed: https://github.com/eslint/espree/issues/212#issuecomment-198783375

**Comment by nzakas** [comment]
Can you open a new one? I'll close this.
